packages/qmongjs/src/components/FilterMenu/TreeViewFilterSection.tsx

```diff
@@ -208,18 +194,77 @@ export const initDefaultExpanded = (
 };
 
 /**
- * Component for use with the FilterMenu component, which uses a TreeView to display filter options.
- * The component can be used for either single or multi-select filters.
+ * Function that returns a function to handle checkbox change events
+ *
+ * @param selectedIds The ids (values) of the currently selected nodes
+ * @param filterKey The key for the filter section
+ * @param isMultiSelect Whether the tree is multi- or single select
+ * @param filterSettingsValuesMap Map for looking up FilterSettingValues by ids
+ * @param filterSettingsDispatch Dispatch function for the filter settings reducer
+ * @returns A handler updating the selected nodes, which take a boolean and a node id (value) as input
+ */
+export const selectionHandlerFunc = (
+  selectedIds: string[],
+  filterKey: string,
+  isMultiSelect: boolean,
+  filterSettingsValuesMap: Map<string, TreeViewFilterSettingsValue>,
+  filterSettingsDispatch: React.Dispatch<FilterSettingsAction>,
+) => {
+  return (checked: boolean, nodeId: string) => {
+    let updatedSelectedIds = selectedIds;
+
+    if (checked) {
+      if (isMultiSelect) {
+        // TODO: Control must support an optional maximum allowed selections.
```

Comment:
Lag issue av denne?

Reply:
Har lagt inn parameter for maksimalt antall man kan velge i slengen. Hvis man forsøker å velge flere, vises en warning-label.